src/views/Deposit/components/EnterAmount/index.tsx

```diff
@@ -17,7 +22,89 @@ import {
 } from './components';
 import { H2, P2 } from 'components/Typography';
 
-export const EnterAmount: React.FC = () => {
+// Constants
+import { TOKEN_GEYSER_ADDRESS, WPOKT_ADDRESS } from 'constants/index';
+
+// Contexts
+import { BalanceContext } from 'contexts/Balance';
+import { Web3Context } from 'contexts/Web3';
+
+// Utils
+import { approve, bnToDec, decToBn, getAllowance, getNotification, stake } from 'utils';
+
+interface IEnterAmount {
+  farmSelected: boolean;
+  setFarmSelected: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const EnterAmount: React.FC<IEnterAmount> = ({ farmSelected, setFarmSelected }) => {
+  const { wpoktBalance } = React.useContext(BalanceContext);
+  const [isApproved, setIsApproved] = React.useState<boolean>(true);
+  const [isDisabled, setIsDisabled] = React.useState<boolean>(true);
+  const [wpoktInputValue, setWpoktInputValue] = React.useState<string>('');
+  const { address, onboard, provider, notify, signer } = React.useContext(Web3Context);
+
+  // Should probably a separated hook
+  const useApproval = async () => {
+    if (address && provider && !isDisabled && farmSelected) {
+      const allowance = await getAllowance(address, TOKEN_GEYSER_ADDRESS, WPOKT_ADDRESS, provider);
+      if (+decToBn(+wpoktInputValue) > +allowance) {
+        setIsApproved(true);
+      } else {
+        setIsApproved(false);
+      }
+    }
+  };
+  useApproval();
+
+  // Effects
+  React.useEffect(() => {
+    if (wpoktInputValue === '' || wpoktInputValue === '0' || !farmSelected) {
+      setIsDisabled(true);
+    } else {
+      setIsDisabled(false);
+    }
+  }, [wpoktInputValue, farmSelected]);
+
+  // Handlers
+  const onDeposit = async () => {
+    onboard?.walletCheck();
+    if (address && provider && !isDisabled && farmSelected) {
+      if (isApproved && signer) {
+        const response = await approve(
+          decToBn(+wpoktInputValue).toString(),
+          TOKEN_GEYSER_ADDRESS,
+          WPOKT_ADDRESS,
+          signer,
+        );
+        if (typeof response === 'boolean') {
+          console.log(response);
+        } else {
+          if (notify) {
+            getNotification(notify, response);
+          }
+        }
+      } else {
+        if (signer) {
+          const response = await stake(decToBn(+wpoktInputValue).toString(), TOKEN_GEYSER_ADDRESS, signer);
+          if (typeof response === 'boolean') {
+            console.log(response);
+          } else {
+            if (notify) {
+              getNotification(notify, response);
+              setWpoktInputValue('');
+              setFarmSelected(false);
+            }
+          }
+        }
+      }
+    }
```

Comment:
In terms of logic, this is a bit hard to follow due to all the nesting... We could probably try to flatten it a bit by inverting the logic somewhat, e.g if any of the values checked on the first if is missing, we execute the first case, else we can just check by case.

Reply:
Fixed in 856de0c7c7a7109633b56c9cc98309dbdf193b48